addons/arsenal/functions/fnc_addVirtualItems.sqf

```diff
@@ -21,206 +21,203 @@
 
+        // Add onto existing items, in case some items that were already added aren't available by default in the arsenal
         {
             (_x select 0) append (_x select 1);
-            (_x select 2) set [(_x select 3), (_x select 0) arrayIntersect (_x select 0)];
+            (_x select 2) set [_x select 3, (_x select 0) arrayIntersect (_x select 0)];
         } forEach [
-            [(_cargo select 0 select 0),(_configItems select 0 select 0), _cargo select 0, 0],
-            [(_cargo select 0 select 1),(_configItems select 0 select 1), _cargo select 0, 1],
-            [(_cargo select 0 select 2),(_configItems select 0 select 2), _cargo select 0, 2],
-            [(_cargo select 1 select 0),(_configItems select 1 select 0), _cargo select 1, 0],
-            [(_cargo select 1 select 1),(_configItems select 1 select 1), _cargo select 1, 1],
-            [(_cargo select 1 select 2),(_configItems select 1 select 2), _cargo select 1, 2],
-            [(_cargo select 1 select 3),(_configItems select 1 select 3), _cargo select 1, 3]
+            [_weapons select IDX_VIRT_PRIMARY_WEAPONS, _configItems select IDX_VIRT_WEAPONS select IDX_VIRT_PRIMARY_WEAPONS, _weapons, IDX_VIRT_PRIMARY_WEAPONS],
+            [_weapons select IDX_VIRT_SECONDARY_WEAPONS, _configItems select IDX_VIRT_WEAPONS select IDX_VIRT_SECONDARY_WEAPONS, _weapons, IDX_VIRT_SECONDARY_WEAPONS],
+            [_weapons select IDX_VIRT_HANDGUN_WEAPONS, _configItems select IDX_VIRT_WEAPONS select IDX_VIRT_HANDGUN_WEAPONS, _weapons, IDX_VIRT_HANDGUN_WEAPONS],
+            [_weaponAttachments select IDX_VIRT_OPTICS_ATTACHMENTS, _configItems select IDX_VIRT_ATTACHMENTS select IDX_VIRT_OPTICS_ATTACHMENTS, _weaponAttachments, IDX_VIRT_OPTICS_ATTACHMENTS],
+            [_weaponAttachments select IDX_VIRT_FLASHLIGHT_ATTACHMENTS, _configItems select IDX_VIRT_ATTACHMENTS select IDX_VIRT_FLASHLIGHT_ATTACHMENTS, _weaponAttachments, IDX_VIRT_FLASHLIGHT_ATTACHMENTS],
+            [_weaponAttachments select IDX_VIRT_MUZZLE_ATTACHMENTS, _configItems select IDX_VIRT_ATTACHMENTS select IDX_VIRT_MUZZLE_ATTACHMENTS, _weaponAttachments, IDX_VIRT_MUZZLE_ATTACHMENTS],
+            [_weaponAttachments select IDX_VIRT_BIPOD_ATTACHMENTS, _configItems select IDX_VIRT_ATTACHMENTS select IDX_VIRT_BIPOD_ATTACHMENTS, _weaponAttachments, IDX_VIRT_BIPOD_ATTACHMENTS]
         ];
 
-        for "_index" from 2 to 17 do {
+        // Add onto existing items, in case some items that were already added aren't available by default in the arsenal
+        for "_index" from IDX_VIRT_ITEMS_ALL to IDX_VIRT_MISC_ITEMS do {
             (_cargo select _index) append (_configItems select _index);
             _cargo set [_index, (_cargo select _index) arrayIntersect (_cargo select _index)];
         };
     };
-
 } else {
+    // Make sure all items are in string form, then convert to config case (non-existent items return "")
+    _items = (_items select {_x isEqualType ""}) apply {_x call EFUNC(common,getConfigName)};
```

Comment:
Do we want to be able to add items that have `scope` < 2? If so, what are the limits? 1? 0? Should `scopeArsenal` be considered here?

Reply:
scope less than 2 should usually not be displayed.
scopeArsenal is good question, what does vanilla arsenal do?

Reply:
`BIS_fnc_addVirtualItemCargo` does not check anything.

In `BIS_fnc_arsenal`, there are two different types of checking:
```SQF
if (isNumber (_x >> "scopeArsenal")) then { getNumber (_x >> "scopeArsenal") } else { getNumber (_x >> "scope") }
```
for weapons, glasses, backpacks and more
and
```SQF
if (getNumber (_cfgMag >> "scope") == 2 || getNumber (_cfgMag >> "scopeArsenal") == 2) then
```
for magazines and weapon attachments.

Reply:
Addressed in [4c9eb8b](https://github.com/acemod/ACE3/pull/9040/commits/4c9eb8b480d2dc32c95f150a8fd040f6cc527af6).